.github/workflows/update-israeli-bank-scrapers.yml

```
@@ -0,0 +1,77 @@
+name: Update israeli-bank-scrapers
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  # Manual trigger
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: false
+        default: 'information'
+        type: choice
+        options:
+          - information
+          - debug
+          - warning
+          - critical
+      tags:
+        description: 'Purpose of Run This Workflow?'
+        required: false
+        type: string
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Get old version
+        id: get_old_version
+        run: |
+          old_version=$(grep 'israeli-bank-scrapers-core@' yarn.lock | head -n1 | cut -d'@' -f2)
+          echo "::set-output name=old_version::$old_version"
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Upgrade israeli-bank-scrapers-core
+        run: |
+          yarn upgrade israeli-bank-scrapers-core --latest
+
+      - name: Get new version
+        id: get_new_version
+        run: |
+          new_version=$(grep 'israeli-bank-scrapers-core@' yarn.lock | cut -d'@' -f2)
+          echo "::set-output name=new_version::$new_version"
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add .
+          git commit -m "chore: Upgrade israeli-bank-scrapers-core library" || echo "No changes to commit"
+
+      - name: Generate diff link
+        id: diff_link
+        run: |
+          echo "::set-output name=diff_link::https://github.com/eshaham/israeli-bank-scrapers/compare/v${{ steps.get_old_version.outputs.old_version }}...v${{ steps.get_new_version.outputs.new_version }}"
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v6
+        with:
```

For security reasons, I review the code of the action to see it is secured, then I lock it into a speicifc commit.


```suggestion
        uses: peter-evans/create-pull-request@b1ddad2c994a25fbc81a28b3ec0e368bb2021c50
```

---

I didn't understand. why not using v6?

---

You can see [here](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management) how the versioning works.

Not only on Github Actions, but in other git-based versioning, we need to be careful when we point to a **tagged** version, because it gives some control to the action publisher.

If a malicious actor gets control of the action, he will add a malicious code into the action and will move the tag to the malicious code.
But he never can change the commit ID.